tests/test_message_meta.py

```diff
@@ -36,25 +32,29 @@ def index_type(request: pytest.FixtureRequest) -> typing.Literal["normal", "skip
 
 
 @pytest.fixture(scope="function")
-def df(df_type: typing.Literal['cudf', 'pandas'],
+def df(_filter_probs_df: cudf.DataFrame,
+       df_type: typing.Literal['cudf', 'pandas'],
        index_type: typing.Literal['normal', 'skip', 'dup', 'down', 'updown'],
        use_cpp: bool):
 
-    loaded_df = read_file_to_df(os.path.join(TEST_DIRS.tests_data_dir, 'filter_probs.csv'),
-                                file_type=FileTypes.Auto,
-                                df_type=df_type)
+    if df_type == 'cudf':
+        loaded_df = _filter_probs_df.copy(deep=True)
+    elif df_type == 'pandas':
+        loaded_df = _filter_probs_df.to_pandas()
+    else:
+        assert False, "Unknown df_type type"
```

Comment:
This should be changed to just `filter_probs_df` which should be parameterized already. Then we dont need any of this conversion code to be repeated

Reply:
The reason why it isn't is that the `_filter_probs_df` fixture is a session scoped fixture, while `filter_probs_df` is a per-function fixture. 

So that we read it once, and make a per-function copy.

Reply:
Understood. We should have a session scoped `_filter_probs_df` and then a function scoped `filter_probs_df` fixture. The `filter_probs_df` should request both `df_type` and `use_cpp` so it gets parameterized on both. Then in `test_message_meta` all we need to do is add the `filter_probs_df` fixture which will handle this code for us.